packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,548 @@
+}
+
+// Declined
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "DECLINED",
+  reason: "Declined by user."
+}
+
+// Pending - Simple
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // will be included in the requests body
+    params: {}, // will be included in the requests url
+  }
+}
+
+// Pending - First Time with Local
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // included in body of request
+    params: {}, // included as query params on endpoint
+  },
+  local: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "view/frame",
+    endpoint: "https://____", // the iframe that will be rendered,
+    data: {}, // sent to frame when ready
+    params: {}, // included as query params on endpoint
+  }
+}
+```
+
+#### data and params
+
+`data` and `params` are information that the wallet can provide in the service config that FCL will pass back to the service.
+- `params` will be added onto the `endpoint` as query params.
+- `data` will be included in the body of the `HTTP/POST` request or in the `FCL:VIEW:READY:RESPONSE` for a `IFRAME/RPC`, `POP/RPC` or `TAB/RPC`.
+
+# Authorization Service
+
+Authorization services are depicted with with a `type: "authz"`, and a `method` of either `HTTP/POST`, `IFRAME/RPC`, `POP/RPC` or `TAB/RPC`.
+They are expected to eventually return a `f_type: "CompositeSignature"`.
+
+An authorization service is expected to know the Account and the Key that will be used to sign the transaction at the time the service is sent to FCL (during authentication).
+
+```javascript
+{
+  f_type: "Service",
+  f_vsn: "1.0.0",
+  type: "authz",               // say its an authorization service
+  uid: "amazing-wallet#authz", // standard service uid
+  method: "HTTP/POST",         // can also be `IFRAME/RPC` or `POP/RPC`
+  endpoint: "https://____",    // where to talk to the service
+  identity: {
+    f_type: "Identity",
+    f_vsn: "1.0.0",
+    address: "0xUser",         // the address that the signature will be for
+    keyId: 0,                  // the key for the address that the signature will be for
+  },
+  data: {},
+  params: {},
+}
+```
+
+FCL will use the `method` provided to request a composite signature from authorization service (Wrapped in a `PollingResponse`).
+The authorization service will be sent a `Signable`.
+The service is expected to construct an encoded message to sign from `Signable.voucher`.
+It then needs to hash the encoded message, tag the message, and then sign it producing a signature.
```

Comment:
Any more detail on what it means to "tag" the message?
It then needs to hash the encoded message, and append a required tag.
Finally it signs the payload with the user/s keys, producing a signature.